jfr-streaming/src/test/java/io/opentelemetry/contrib/jfr/metrics/GenerateDocs.java

```diff
@@ -0,0 +1,150 @@
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class GenerateDocs {
+
+  private static final Logger LOGGER = Logger.getLogger(GenerateDocs.class.getName());
+
+  private static final String JFR_README_PATH_KEY = "jfr.readme.path";
+  private static final String START = "<!-- generateDocsStart -->";
+  private static final String END = "<!-- generateDocsEnd -->";
+  private static final Pattern PATTERN = Pattern.compile(START + ".*" + END, Pattern.DOTALL);
+
+  private GenerateDocs() {}
+
+  public static void main(String[] args) throws Exception {
+    // Suppress info level logs
+    Logger.getLogger(JfrTelemetry.class.getName()).setLevel(Level.WARNING);
+
+    String jfrReadmePath = System.getProperty(JFR_README_PATH_KEY);
+    if (jfrReadmePath == null) {
+      throw new IllegalStateException(JFR_README_PATH_KEY + " is required");
+    }
+
+    LOGGER.info("Generating JFR docs. Writing to " + jfrReadmePath);
+    String markdownTable = generateMarkdownTable();
+    LOGGER.info("Markdown table: " + System.lineSeparator() + markdownTable);
+    writeReadme(markdownTable, jfrReadmePath);
+    LOGGER.info("Done");
+  }
+
+  private static String generateMarkdownTable() throws InterruptedException {
+    Map<JfrFeature, String> jfrFeatureRows = new ConcurrentHashMap<>();
+    List<Thread> threads = new ArrayList<>();
+    for (JfrFeature feature : JfrFeature.values()) {
+      Thread thread =
+          new Thread(
+              () -> {
+                try {
+                  String metricCol =
+                      metricsForJfrFeature(feature).stream()
+                          .sorted()
+                          .map(s -> "`" + s + "`")
+                          .collect(joining(", "));
+                  jfrFeatureRows.put(
+                      feature,
+                      "| "
+                          + feature.name()
+                          + " | "
+                          + feature.isDefaultEnabled()
+                          + " | "
+                          + metricCol
+                          + " |"
+                          + System.lineSeparator());
+                } catch (InterruptedException e) {
+                  Thread.currentThread().interrupt();
+                }
+              });
+      threads.add(thread);
+      thread.start();
+    }
+
+    for (Thread thread : threads) {
```

Comment:
I think it's marginally cleaner to use an `ExecutorService` for this stuff, but no biggie.

Reply:
Realized I don't need to do these in separate threads at all. Can just start a bunch of JfrTelemetry instances in the current thread. They'll each subscribe to independent event streams and record only metrics for a particular JfrFeature.